packages/@lwc/engine-core/src/framework/modules/computed-style-attr.ts

```diff
@@ -14,17 +14,20 @@ function updateStyleAttribute(oldVnode: VNode, vnode: VNode) {
         data: { style: newStyle },
         owner: { renderer },
     } = vnode;
-    const { getStyleDeclaration, removeAttribute } = renderer;
+    const { setCSSStyleProperty, removeAttribute } = renderer;
     if (oldVnode.data.style === newStyle) {
         return;
     }
 
-    const style = getStyleDeclaration(elm);
+    removeAttribute(elm, 'style');
```

Comment:
Because `setCSSStyleProperty` always adds, when updating the whole style attribute, it needs to be cleared out first.